trie/trie.go

```diff
@@ -187,3 +292,11 @@ func lenCommonPrefix(a, b []byte) int {
 
 	return length
 }
+
+// concat concatenates two byte arrays
+func concat(s1 []byte, s2 ...byte) []byte {
+	r := make([]byte, len(s1)+len(s2))
+	copy(r, s1)
+	copy(r[len(s1):], s2)
+	return r
+}
```

Comment:
`func concat` could be placed in a `common/common.go` file as it could be useful elsewhere and not only `trie.go`

Reply:
good move, will do